newrelic-agent/src/main/java/com/newrelic/agent/stats/ResponseTimeStatsImpl.java

```diff
@@ -67,94 +71,116 @@ public void recordResponseTimeInNanos(long responseTime) {
     public void recordResponseTimeInNanos(long responseTime, long exclusiveTime) {
         double responseTimeAsDouble = responseTime;
         responseTimeAsDouble *= responseTimeAsDouble;
-        sumOfSquares += responseTimeAsDouble;
-        if (count > 0) {
-            minValue = Math.min(responseTime, minValue);
-        } else {
-            minValue = responseTime;
-        }
-        count++;
-        total += responseTime;
-        maxValue = Math.max(responseTime, maxValue);
-        totalExclusive += exclusiveTime;
-        if (NewRelic.getAgent().getConfig().getValue(AgentConfigImpl.METRIC_DEBUG, AgentConfigImpl.DEFAULT_METRIC_DEBUG)) {
-            if (count < 0 || total < 0 || totalExclusive < 0 || sumOfSquares < 0) {
-                NewRelic.incrementCounter("Supportability/ResponseTimeStatsImpl/NegativeValue");
-                Agent.LOG.log(Level.INFO, "Invalid count {0}, total {1}, totalExclusive {2}, or sum of squares {3}",
-                        count, total, totalExclusive, sumOfSquares);
+        synchronized (lock) {
+            sumOfSquares += responseTimeAsDouble;
+            if (count > 0) {
+                minValue = Math.min(responseTime, minValue);
+            } else {
+                minValue = responseTime;
+            }
+            count++;
+            total += responseTime;
+            maxValue = Math.max(responseTime, maxValue);
+            totalExclusive += exclusiveTime;
+            if (NewRelic.getAgent().getConfig().getValue(AgentConfigImpl.METRIC_DEBUG, AgentConfigImpl.DEFAULT_METRIC_DEBUG)) {
+                if (count < 0 || total < 0 || totalExclusive < 0 || sumOfSquares < 0) {
+                    NewRelic.incrementCounter("Supportability/ResponseTimeStatsImpl/NegativeValue");
+                    Agent.LOG.log(Level.INFO, "Invalid count {0}, total {1}, totalExclusive {2}, or sum of squares {3}",
+                            count, total, totalExclusive, sumOfSquares);
+                }
             }
         }
-
     }
 
     @Override
     public boolean hasData() {
-        return count > 0 || total > 0 || totalExclusive > 0;
+        boolean hasData;
+        synchronized (lock) {
+            hasData = count > 0 || total > 0 || totalExclusive > 0;
```

Comment:
possibly fewer branches: `(count | total | totalExclusive) != 0`.  _Probably_ violates your code style, I'd expect.  

this is also cacheable. in all your mutators you could flip a stored `hasData` field and since booleans are atomic, you could mark `volatile` and avoid a lock.  Micro-optimization and I'd only consider if this were in critical piece of code

Reply:
cool...yeah this is existing logic so I was avoiding changing it but might be worth it...I was just adding `synchronize` around things that were being changed